### PR TITLE
[Public collections] Add sorting to Asset endpoint

### LINKED ISSUE #2538: Fix public collections endpoint after asset_type:collection change
## Description

The endpoint `/api/v2/assets/?all_public=true` (old `/api/v2/collections/?all_public=true`) should return a list of public collections. 

## Additional details

Requirement for #2293

### LINKED ISSUE #2539: Assets endpoint sorting
## Description

`/api/v2/assets/` should allow for sorting by:
- `settings.country`
- `asset_type`
- `settings.languages`
- `date_modified`
- `name`
- `settings.organization`
- `owner__username`
- `settings.sector`
- `subscribers_count`

Could be `sort={"owner__username":-1}` to be consistent with submissions endpoint, but it's not a fixed requirement.

## Additional details

Requirement for #2293

### PR DESCRIPTION
**Pull request by noliveleger**
It's possible to sort by these fields: 

- `asset_type`
- `date_modified`
- `name`
- `owner__username`
- `subscribers_count`

Django default parameter is `ordering` but it can be changed to match `submissions` ordering parameter (i.e. `sort`)

It's also possible to filter by these fields: 
- `settings__country__label`
- `settings__organization`
- `settings__sector_label`
- `summary__languages` 

But also by `status`
- `/api/v2/assets?status=private&q=asset_type:collection` -> collections owned by user
- `/api/v2/assets?status=shared&q=asset_type:collection` -> collections user can view
- `/api/v2/assets?status=public&q=asset_type:collection` -> collections user has subscribed to (ONLY)
- `/api/v2/assets?status=public-discoverable&q=asset_type:collection` -> all public collections

Names have been chosen to respect Django ORM (and Whoosh?) nomenclature (but could be changed) 

## NOT RETRO-COMPATIBLE CHANGE
Before, to get assets without no parents, `parent=` should be passed among querystring parameters.
It is not the case anymore:
To (try to) be consistent, it has to be passed as `parent:null` inside `q` parameter. e.g: `q=parent:null` (see #2575)

## EXAMPLE: 
to retrieve assets with no parents, with type: `collection, block, template` and with languages: `English (en)` ordered by subscribers_count (desc) , request should like this 

`https://kpi/api/v2/assets/?q=parent:null AND asset_type:collection OR asset_type:block OR asset_type:template AND summary__languages__contains:"English (en)"&ordering=-subscribers_count`

Languages can be case-insensitive by using : `summary__languages__icontains:"English (en)"`

Closes #2538 
Closes #2539

### CODE REVIEW

**Comment by noliveleger** [comment]
Search by `summary__language` does not work anymore.
Error:
`Unsupported lookup 'languages' for JSONField or join on the field not permitted`

May be fixed by #2635

**Review by jnm — CHANGES_REQUESTED** [review]

File: kpi/filters.py [hunk 1]
```diff
@@ -39,63 +63,134 @@ def filter_queryset(self, request, queryset, view):
...
+        STATUS_PARAMETER = 'status'
+
+        try:
+            q = request.query_params['q'].strip()
+            if q == '':
+                return queryset
+        except KeyError:
+            return queryset
+
+        query_parts = q.split(' AND ')  # Can be risky if one of values contains ` AND `
```
[the remark this answers is not in the document]
Comment (jnm) [reply kpi/filters.py:134]: we've opted to add `status` as its own query parameter (and allow filtering only by one `status` at a time)

File: kpi/views/v2/asset.py [hunk 2]
```diff
@@ -216,6 +380,93 @@ def get_queryset(self, *args, **kwargs):
...
+        # Languages
+        records.query.clear_ordering(True)
+
+        for record in records.all():
+            try:
+                for language in record['summary']['languages']:
+                    if language:
+                        metadata['languages'].add(language)
+            except (ValueError, KeyError):
+                pass
```
[the remark this answers is not in the document]
Comment (jnm) [reply kpi/views/v2/asset.py:418]: wrong! `ValueError` is to guard against `None['oops']`

File: kpi/utils/asset_content_analyzer.py [hunk 3]
```diff
@@ -61,3 +62,15 @@ def get_summary(self):
...
+
+    @staticmethod
+    def format_translations(translations):
+        if not translations:
+            return translations
+
+        formatted_translations = []
+        for translation in translations:
+            if translation is not None:
+                formatted_translations.append(translation.capitalize())
```
[the remark this answers is not in the document]
Comment (jnm) [reply kpi/utils/asset_content_analyzer.py:74]: we've removed `format_translations()` :cry:

**Review by jnm — APPROVED** [review]

**Pull request merged**